src/FormControl.js

```diff
@@ -62,6 +64,13 @@ class FormControl extends React.Component {
       classes = getClassSet(bsProps);
     }
 
+    // If user provides a size, make sure to append it to classes as input-
+    // e.g. if bsSize is small, it will append input-sm
+    if (bsSize) {
+      const size = SIZE_MAP[bsSize] || bsSize;
+      classes[`input-${size}`] = true;
```

Comment:
@taion initially I was planning to but then `prefix` uses the bsClass in prop and I thought it might just be quicker to add this one line instead. I'll change it anyway :)